toolkit/tools/scheduler/schedulerutils/depsolver.go

```diff
@@ -40,6 +48,22 @@ func CanSubGraph(pkgGraph *pkggraph.PkgGraph, node *pkggraph.PkgNode, useCachedI
 
 		// This node is not yet solvable
 		logger.Log.Warnf("Could not subgraph due to node: %v", pkgNode)
+
+		// If we are in trace mode, print the path from the root node to the unsolvable node
+		if logger.Log.IsLevelEnabled(logrus.TraceLevel) {
+			pt, ok := path.BellmanFordFrom(node, pkgGraph)
+			if !ok {
+				logger.Log.Warnf("Could not run BellmanFordFrom on %v", node)
+				return
+			} else {
+				path, _ := pt.To(pkgNode.ID())
+				logger.Log.Tracef("Path between %v and %v:", node, pkgNode)
+				for _, n := range path {
+					logger.Log.Warnf("  %v", n.(*pkggraph.PkgNode))
```

Comment:
Why log the nodes on the warning level if we run all of this only if we're logging on trace level? I understand the failure above, but this seems like something we expect to log.

Reply:
Ah, forgot to commit the switch to trace.